### PR TITLE
feat: add shell completions

### PR DESCRIPTION
**Pull request by ryanccn**
This PR adds completions to the CLI using [clap_complete](https://docs.rs/clap_complete/latest/clap_complete/) and also installs the completions in the Nix package.

### CODE REVIEW

File: src/cli/cmd/mod.rs
Comment (lucperkins) [inline src/cli/cmd/mod.rs:35]: kubectl, docker, hugo, and a lot of others use `<tool> completion <shell>` as their format. I'd prefer to stick with that format and make this `fh completion` instead.

**Review by lucperkins — CHANGES_REQUESTED** [review]

**Comment by lucperkins** [comment]
LGTM but I'm going to wait on a +1 from @cole-h

**Review by cole-h — APPROVED** [review]
LGTM, seems to work fine (when manually sourcing it in fish, at least).

**Pull request merged**
